backend/controllers/prisonerProfile/prisonerProfileContext.ts

```diff
@@ -1,26 +1,33 @@
 const getContext = async ({ offenderNo, res, req, oauthApi, systemOauthClient, restrictedPatientApi }) => {
   if (res.locals.user === undefined) {
-    return { context: res.locals, overrideAccess: false }
+    return { context: res.locals, restrictedPatientDetails: undefined }
   }
   const {
     user: { allCaseloads },
   } = res.locals
 
   const { username } = req.session.userDetails
   const userRoles = oauthApi.userRoles(res.locals)
+  const isPom = userRoles.map((role) => role.roleCode).includes('POM')
 
-  if (userRoles.map((userRole) => userRole.roleCode).includes('POM')) {
-    const isRestrictedPatient = await restrictedPatientApi.isCaseLoadRestrictedPatient(
-      res.locals,
-      offenderNo,
-      allCaseloads.map((caseload) => caseload.caseLoadId)
-    )
-    if (isRestrictedPatient) {
-      const context = await systemOauthClient.getClientCredentialsTokens(username)
-      return { context, overrideAccess: true }
+  const restrictedPatient = await restrictedPatientApi.getRestrictedPatientDetails(
```

Comment:
doesn't this mean we're now calling the restricted patient api for every single prisoner? That's quite a big increase in the number of calls. Previously we only used to call it if the user was a POM.